packages/arb-token-bridge-ui/tests/e2e/specs/depositCctp.cy.ts

```diff
@@ -3,236 +3,273 @@
+          name: /Continue/i
+        })
+          .should('be.visible')
+          .should('be.enabled')
+          .click()
 
-          // Checkbox
-          cy.findByRole('switch', {
-            name: /I understand that I'll have to send/i
-          })
-            .should('be.visible')
-            .click()
-          cy.findByRole('switch', {
-            name: /I understand that it will take/i
-          })
-            .should('be.visible')
-            .click()
+        cy.findByText(/I understand that I have to/).click()
+        cy.findByRole('button', {
+          name: /Pay approval fee of/
+        }).click()
+        cy.log('Approving USDC...')
+        cy.confirmMetamaskPermissionToSpend(USDCAmountToSend.toString()).then(
+          () => {
+            // eslint-disable-next-line
+            cy.wait(40_000)
+            cy.confirmMetamaskTransaction().then(() => {
+              cy.findByText('Pending transactions').should('be.visible') // tx history should be opened
+              cy.findByText(
+                `${formatAmount(USDCAmountToSend, {
+                  symbol: 'USDC'
+                })}`
+              ).should('be.visible')
+            })
+          }
+        )
+      })
+    })
 
-          cy.findByRole('switch', {
-            name: /I understand USDC.e/i
-          })
-            .should('be.visible')
-            .click()
+    it('should initiate depositing USDC to custom destination address through CCTP successfully', () => {
+      context('should fill custom destination address successfully', () => {
+        // click on advanced settings
+        cy.findByLabelText('advanced settings').should('be.visible').click()
 
-          cy.findByRole('button', {
-            name: 'Confirm'
-          })
-            .should('be.visible')
-            .should('be.enabled')
-            .click()
+        // unlock custom destination address input
+        cy.findByLabelText('Custom destination input lock')
+          .should('be.visible')
+          .click()
 
-          cy.findByText(/pay a one-time approval fee/).click()
-          cy.findByRole('button', {
-            name: /Pay approval fee of/
-          }).click()
-          cy.log('Approving USDC...')
-          cy.confirmMetamaskPermissionToSpend(USDCAmountToSend.toString()).then(
-            () => {
-              // eslint-disable-next-line
-              cy.wait(40_000)
-              cy.confirmMetamaskTransaction().then(() => {
-                cy.findByText('Pending transactions').should('be.visible') // tx history should be opened
-                cy.findByText(
-                  `${formatAmount(USDCAmountToSend, {
-                    symbol: 'USDC'
-                  })}`
-                ).should('be.visible')
-              })
-            }
-          )
+        cy.findByPlaceholderText(Cypress.env('ADDRESS'))
+          .typeRecursively(Cypress.env('CUSTOM_DESTINATION_ADDRESS'))
+          .should('be.visible')
```

Comment:
I think we should check for visibility before typing?

Reply:
Done